src/Spago/Command/Uninstall.purs

```diff
@@ -0,0 +1,129 @@
+            , modifyConfig: \newDeps packageConfig ->
+                packageConfig { dependencies = newDeps }
+            }
+  missingTestConfigOrContext <- case workspace.selected of
+    Just p ->
+      pure $ toContext (Path.concat [ p.path, "spago.yaml" ]) p.package
+    Nothing -> do
+      case workspace.rootPackage of
+        Nothing ->
+          die "No package was selected. Please select a package."
+        Just p ->
+          pure $ toContext "spago.yaml" p
+  case missingTestConfigOrContext of
+    Left pkgName ->
+      logWarn $ "Could not uninstall test dependencies for " <> PackageName.print pkgName <> " because it does not have a test configuration."
+    Right context -> do
+      logDebug $ "Existing " <> context.sourceOrTestString <> " dependencies are: " <> (Array.intercalate ", " $ foldlWithIndex (\k a _ -> Array.snoc a $ PackageName.print k) [] $ unwrap context.deps)
+      let
+        { warn, removed, newDeps } = foldl deleteOrWarn init args.dependenciesToRemove
+          where
+          init = { warn: [], removed: [], newDeps: unwrap context.deps }
+          deleteOrWarn acc next
+            | Just (Tuple _ newDeps) <- Map.pop next acc.newDeps = acc { newDeps = newDeps, removed = Array.snoc acc.removed next }
+            | otherwise = acc { warn = Array.snoc acc.warn next }
+      for_ (NEA.fromArray warn) \undeclaredPkgs ->
+        logWarn
+          [ "The following packages cannot be uninstalled because they are not declared in the package's " <> context.sourceOrTestString <> " dependencies:"
+          , "  " <> NEA.intercalate ", " (map PackageName.print undeclaredPkgs)
+          ]
+
+      case NEA.fromArray removed of
+        Nothing ->
+          logInfo $ "The package config for " <> PackageName.print context.name <> " was not updated."
+        Just removed' -> do
+          logInfo
+            [ "Removing the following " <> context.sourceOrTestString <> " dependencies:"
+            , "  " <> NEA.intercalate ", " (map PackageName.print removed')
+            ]
+          logDebug $ "Editing config file at path: " <> context.configPath
+          editSpagoYaml
+            { configPath: context.configPath
+            , onError: \err ->
+                die $ "Error decoding package config for package " <> PackageName.print context.name <> ":\n" <> err
+            , modifyConfig: \config ->
+                config { package = map (context.modifyConfig (Dependencies newDeps)) config.package }
+            }
+
+editSpagoYaml
+  :: forall m
+   . MonadAff m
+  => { configPath :: FilePath
+     , modifyConfig :: Core.Config -> Core.Config
+     , onError :: String -> m Unit
+     }
+  -> m Unit
+editSpagoYaml { configPath, modifyConfig, onError } = do
+  content <- liftAff $ FS.readYamlDocFile Core.configCodec configPath
+  case content of
+    Left err ->
+      onError err
+    Right { yaml: config } ->
+      liftAff $ FS.writeYamlFile Core.configCodec configPath $ modifyConfig config
```

Comment:
This piece of code was fine for tests, but it's not good to use for normal business - rewriting the config like that loses comments and extra fields the config might contain.

We should instead manipulate the `Yaml.Doc`, and use the same setup [as it's done in the Config module](https://github.com/purescript/spago/blob/23697f4f7e93cfcdf7d71b0833f9591588747948/src/Spago/Config.purs#L521), where we get the `Doc` from the workspace, and apply an FFI function that does the yaml manipulation while preserving comments.

Reply:
Ah, ok. Let me fix that.